pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -148,21 +148,24 @@ func (pa *PodAutoscaler) PanicThresholdPercentage() (percentage float64, ok bool
 
 // IsReady looks at the conditions and if the Status has a condition
 // PodAutoscalerConditionReady returns true if ConditionStatus is True
-func (pas *PodAutoscalerStatus) IsReady() bool {
-	return podCondSet.Manage(pas).IsHappy()
+func (pa *PodAutoscaler) IsReady() bool {
+	pas := pa.Status
+	return pa.Generation == pas.ObservedGeneration && podCondSet.Manage(&pas).IsHappy()
 }
 
 // IsActivating returns true if the pod autoscaler is Activating if it is neither
 // Active nor Inactive
-func (pas *PodAutoscalerStatus) IsActivating() bool {
-	cond := pas.GetCondition(PodAutoscalerConditionActive)
-	return cond != nil && cond.Status == corev1.ConditionUnknown
+func (pa *PodAutoscaler) IsActivating() bool {
+	pas := pa.Status
+	return pa.Generation != pas.ObservedGeneration ||
+		pas.GetCondition(PodAutoscalerConditionActive).IsUnknown()
 }
 
 // IsInactive returns true if the pod autoscaler is Inactive.
-func (pas *PodAutoscalerStatus) IsInactive() bool {
-	cond := pas.GetCondition(PodAutoscalerConditionActive)
-	return cond != nil && cond.Status == corev1.ConditionFalse
+func (pa *PodAutoscaler) IsInactive() bool {
+	pas := pa.Status
+	return pa.Generation == pas.ObservedGeneration &&
```

Comment:
Currently, our active condition is inferred by `IsInactive()`. So does this mean now active means `(pa.Generation == pas.ObservedGeneration && !pas.GetCondition(PodAutoscalerConditionActive).IsFalse()) || (pa.Generation != pas.ObservedGeneration && !pas.GetCondition(PodAutoscalerConditionActive).IsFalse())`? The latter doesn't mean active, right?

Reply:
Active means pa.Generation == pas.ObservedGeneration && pas.GetCondition(PodAutoscalerConditionActive).IsTrue()

Active should not be inferred by !InActive which is ambiguously either False or Unknown.

Reply:
Right it shouldn't be inferred by `IsInactive()`. There's this piece of logic https://github.com/knative/serving/blob/master/pkg/reconciler/autoscaling/kpa/kpa.go#L270 would this mark PA active when the generations are not matching?

Reply:
Part of the weirdness here is that this reconciler is using its own Status as input rather than Spec+State of the World. I think we still want to consider gen mismatch as unknown, but let's revert that for a different PR for discussion.